crates/node/src/service.rs

```diff
@@ -586,30 +591,42 @@ where
         Ok(timestamp)
     };
 
-    let manual_seal = match sealing {
-        Sealing::Manual => future::Either::Left(sc_consensus_manual_seal::run_manual_seal(
-            sc_consensus_manual_seal::ManualSealParams {
+    let manual_seal: BoxFuture<_> = match sealing {
+        SealingMode::Manual => {
+            Box::pin(sc_consensus_manual_seal::run_manual_seal(sc_consensus_manual_seal::ManualSealParams {
                 block_import,
                 env: proposer_factory,
                 client,
                 pool: transaction_pool,
-                commands_stream,
+                commands_stream: commands_stream.expect("Manual sealing requires a channel from RPC."),
                 select_chain,
                 consensus_data_provider: None,
                 create_inherent_data_providers,
-            },
-        )),
-        Sealing::Instant => future::Either::Right(sc_consensus_manual_seal::run_instant_seal(
-            sc_consensus_manual_seal::InstantSealParams {
+            }))
+        }
+        SealingMode::Instant { finalize } if finalize => Box::pin(
+            sc_consensus_manual_seal::run_instant_seal_and_finalize(sc_consensus_manual_seal::InstantSealParams {
                 block_import,
                 env: proposer_factory,
                 client,
                 pool: transaction_pool,
                 select_chain,
                 consensus_data_provider: None,
                 create_inherent_data_providers,
-            },
-        )),
+            }),
+        ),
+        SealingMode::Instant { finalize } if !finalize => {
+            Box::pin(sc_consensus_manual_seal::run_instant_seal(sc_consensus_manual_seal::InstantSealParams {
+                block_import,
+                env: proposer_factory,
+                client,
+                pool: transaction_pool,
+                select_chain,
+                consensus_data_provider: None,
+                create_inherent_data_providers,
+            }))
+        }
```

Comment:
For readability, it seems better to
```
SealingMode::Instant { finalize } => {
   let instant_seal_params = sc_consensus_manual_seal::InstantSealParams {
         block_import,
         env: proposer_factory,
         client,
         pool: transaction_pool,
         select_chain,
         consensus_data_provider: None,
         create_inherent_data_providers,
   })
   if (finalize) {
      Box::pin(sc_consensus_manual_seal::run_instant_seal_and_finalize(instant_seal_params)
   } else {
      Box::pin(sc_consensus_manual_seal::run_instant_seal(instant_seal_params)

   }
}
```

wdyt?

Reply:
I don't like nesting `if` inside `match`, but removing code duplication is worth it.

Reply:
usually agree, but this case seems that the alternative is worse